.github/workflows/main.yml

```diff
@@ -0,0 +1,102 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  java:
+    name: Java/Gradle
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        java-version: [11]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+
+      # Special handling to test against Java 8:
+      # antlr version 4.10 requires Java 11 to run the antlr tool, but tests still need to run
+      # against Java 8 - that's being done here:
+      - name: Set up Java 8
+        if: ${{ matrix.java-version == '11' }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 8
+      - name: Configure env with Java 8 home
+        if: ${{ matrix.java-version == '11' }}
+        run:
+          echo "JAVA_8_HOME=$JAVA_HOME" >> $GITHUB_ENV
+      - name: Enable Java 8 tests
+        if: ${{ matrix.java-version == '11' }}
+        run:
+          echo "ADDITIONAL_GRADLE_OPTS=-PalsoTestAgainstJava8" >> $GITHUB_ENV
+      - name: Configure Java 8 home for Gradle
+        if: ${{ matrix.java-version == '11' }}
+        run: |
+          mkdir -p $HOME/.gradle
+          echo "org.gradle.java.installations.fromEnv=JAVA_8_HOME" >> $HOME/.gradle/gradle.properties
+
+      - name: Set up JDK ${{ matrix.java-version }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
```

Comment:
lets prefer `temurin` here (allegedly these are already present on the runners and dont need to be installed)